virtcontainers/kata_agent.go

```diff
@@ -1706,6 +1718,23 @@ func (k *kataAgent) handlePidNamespace(grpcSpec *grpc.Spec, sandbox *Sandbox) bo
 	return sharedPidNs
 }
 
+// handlePidNamespace checks if Pid namespace for a container needs to be shared with its sandbox
+// pid namespace. This function also modifies the grpc spec to remove the pid namespace
+// from the list of namespaces passed to the agent.
+func (k *kataAgent) checkAgentPidNs(container *Container) bool {
+	agentPidNs := false
+
+	for _, env := range container.config.Cmd.Envs {
+		if env.Var == agentPidEnv {
+			if val, err := strconv.ParseBool(env.Value); err == nil {
+				agentPidNs = val
```

Comment:
Should the code break out of the loop ?

Reply:
Could have, Although not very common to define the same environment variable more than once, I want to make sure we take the value of the last definition.

Reply:
I don't think we should depend on the order of traversal over container.config.Cmd.Envs.
It would be better if we have clear policy on which of the multiple environment variables to use.